content/fast/incorporate-performance-budgets-into-your-build-tools/codelab-setting-performance-budgets-with-webpack.md

```diff
@@ -0,0 +1,193 @@
+too large and affect how long your application takes to load. Webpack supports
+setting [performance budgets](https://web.dev/fast/performance-budgets-101)
+based on **asset size** and it can keep an eye on bundle sizes for you.
+
+To see it in action, here’s an app that counts the days left until New Year’s.
+It’s built with [React](https://reactjs.org/) and [moment.js](https://momentjs.com/).
+(Just like real-world apps that increasingly rely on frameworks and libraries. 😉)
+
+![An app that counts the days left until New Year’s day](countdown-app.png)
+
+## Measure
+
+This Glitch already contains the app bundled with webpack.
+
+1. To start, click the Remix to Edit button to make the project editable.
+
+<web-screenshot type="remix"></web-screenshot>
+
+2. Click the **Tools** button.
+
+![Tools button](glitch-tools.png)
+
+3. Then click the **Console button**. It will open the Console in another tab.
+
+<web-screenshot type="console"></web-screenshot>
+
+4. To get a color-coded list of assets and their sizes, in the Glitch console
+   type:
+
+<pre class="devsite-terminal devsite-click-to-copy">webpack</pre>
+
+The main bundle is highlighted in yellow because it’s larger than 244 KiB (250
+KB).
+
+<div class="aside note">
+    Webpack uses the unit "KiB", which you might not have 
+    seen before. 1 “KiB” and 1 “KB” are fairly close in size: 1 KiB is 1024 bytes, 
+    while 1 KB is 1000 bytes.)
+</div>
+
+![Webpack output showing bundle size of 323 KiB](webpack-bundle-sizes.png)
+
+These warnings are enabled by default in [production mode](https://webpack.js.org/concepts/mode/)
+and the default threshold is **244 KiB uncompressed**, for both assets and
+[entry points](https://webpack.js.org/concepts/entry-points/)
+(the combination of all assets used during the initial load of a page).
+
+<figure>
+  <img src="./webpack-warning.png" alt="Webpack warning that the asset exceeds the recommended size limit">
+  <figcaption>
+    Webpack warning you about bulky JS bundle ⚠️
+  </figcaption>
+</figure>
+
+Webpack will not only warn you, but it will also give you a recommendation on
+how to downsize your bundles. You can learn more about the recommended techniques on
+[Web Fundamentals](https://developers.google.com/web/fundamentals/performance/webpack/use-long-term-caching#lazy-loading).
+
+<figure> 
+    <img src="webpack-performance-recommendation.png" alt="Webpack performance optimization recommendation" class="screenshot"> 
+    <figcaption>Webpack performance optimization recommendation 💁</figcaption> 
+</figure>
+
+## Set a custom performance budget
+
+An appropriate performance budget will depend on the nature of your project.
+It’s always best to [do your own research](https://web.dev/fast/your-first-performance-budget).
+A [good rule](https://web.dev/fast/your-first-performance-budget#budget-for-quantity-based-metrics)
+of thumb is to deliver under 170 KB of compressed/minified
```

Comment:
would it be worth showing the KiB value as well?

Reply:
I don't think it's relevant here, 170 KB is the default number that we are using in all other guides/recommendations so I'd just let that sink in. People won't have to do KiB mental math when they define webpack performance budgets because the settings are in bytes. The actual numbers from the codelab are shown with equivalents that webpack displays in the warnings + there's the side note.